### PR TITLE
Fix CSS entity output to include trailing space

### PR DESCRIPTION
**Pull request by lunelson**
Without the trailing space, the entity may be interpreted as a higher-range unicode character, depending on what follows. e.g.:

```js
replaceBy('css')('über'); // outputs ​​​​​\00fcber
```
```css
.test {
	content: '\00fcber'; /* does not output über! */
}
```
... the output character is probably one of these http://jrgraphix.net/r/Unicode/

### CODE REVIEW

**Comment by lunelson** [comment]
@m90 Moreover, you should consider encoding characters that have a `charCodeAt(0)` of less than `32`, e.g. newline and tab characters do not otherwise come through correctly

**Comment by m90** [comment]
I'm sorry, but I haven't used this in ages, so I'm slightly out of context.

Is this the same as: https://github.com/m90/entity-convert/pull/1 ?

Regarding your second comment, would you mind either creating an issue or a 2nd PR for that (or include it in this PR even).

**Comment by lunelson** [comment]
I wasn't aware of the issue from #1, I'm not sure it's necessary. I'll do some more digging and then add to this PR

**Comment by lunelson** [comment]
It seems both #1 and this PR are trying to address the same issue, if [this SO post](https://stackoverflow.com/a/8595802) is correct. Essentially, adding the space as I have done, should have the same effect as making sure the code has six characters in total. This is because there is otherwise no terminator character, such as with HTML entities that all end with `;`

**Comment by lunelson** [comment]
Also I'm not sure I understand how your tests 3 ('should leave wrapping html untouched') and 6 ('should leave css selectors untouched') are supposed to work. Your code doesn't contain anything about specifically parsing out wrapping HTML or CSS selectors, as far as I can tell...?

Also: escaping for CSS identifiers follows a different set of rules, best accomplished by this library I believe: https://www.npmjs.com/package/cssesc

**Comment by m90** [comment]
Thanks for finding out, I think I prefer the whitespace solution (i.e. this PR) in that case as it's a whole lot easier to get what's happening. Could you adjust the tests so they pass, so we can get this merged (Travis works again on master as of #3 )?

Regarding the tests: These are very basic and just to make sure that no character like `{` or `<` is being affected by the encoding (which atm does work without any extra parsing on top). This could of course be done way more elegantly.

Regarding all other issues your are mentioning: if you need these features/fixes and want to work on them I would be happy to review PRs or give you push access to the repo, yet I personally will not find the time to tackle these. If you don't want to work on them, but feel they are noteworthy you can also open issues and I'll tag them "help wanted".

**Comment by lunelson** [comment]
I've commented out tests 3 and 6 because I think they are misleading, this code is not specifically handling those conditions in any way, the test just happens to pass; and as I mentioned, escaping CSS identifiers (e.g. class names) is altogether different, so the use-case of passing in a bunch of CSS to be escaped by this code is perhaps unrealistic. In any case, with the right escape code one should be able to display CSS *inside* a CSS `content` rule 🤷‍♂️ 

So to this PR I've just added some commits to make the escaping for purposes of the CSS `content` rule work correctly

File: test.js [hunk 1]
```diff
@@ -15,18 +15,18 @@ describe('entityconvert', function(){
 				, 'abcdefghijklmnopqrstuvwxyz'
 			);
 		});
-		it('should leave wrapping html untouched', function(){
-			assert.equal(
-				ec.html('<span class="foo" data-foo>Ümläut</span>')
-				, '<span class="foo" data-foo>&#220;ml&#228;ut</span>'
-			);
-		});
+		// it('should leave wrapping html untouched', function(){
```
Comment (m90) [inline test.js]: I think I'd like to keep these. Will they still pass? Maybe it's about changing the description?

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
 *.sublime-project
 *.sublime-workspace
 node_modules/
+/package-lock.json
```
Comment (m90) [inline .gitignore:4]: +1, Ideally we'd add a `.npmrc` to the project that says `package-lock=false`, but I can also do that post merge.

File: test.js (hunk 1 above)
Comment (lunelson) [reply test.js]: OK, I've restored those tests. They do currently pass but IMO the use-case could fail under certain conditions, it's hard to predict

Comment (m90) [reply test.js]: I'll try to come up with a less misleading description. Thanks!

**Comment by m90** [comment]
Published to npm as `1.0.0`. Thanks :tada:

**Pull request merged**